central/sensor/service/connection/sensorevents.go

```diff
@@ -106,11 +107,28 @@ func (s *sensorEventHandler) addMultiplexed(ctx context.Context, msg *central.Ms
 		// Node and NodeInventory dedupe on Node ID. We use a different dedupe key for
 		// NodeInventory because the two should not dedupe between themselves.
 		msg.DedupeKey = fmt.Sprintf("NodeInventory:%s", msg.GetDedupeKey())
+	case *central.SensorEvent_ComplianceOperatorResultV2:
+		if !features.ComplianceEnhancements.Enabled() {
+			log.Errorf("Received next gen compliance event from cluster %s (%s). Next gen compliance is disabled on central.", s.cluster.GetName(), s.cluster.GetId())
+			return
+		}
+		// Due to needing both V1 and V2 compliance to run at the same time and due to how the
+		// reconciliation keys are used we need to use the V1 key for reconciliation.  This could
+		// have been avoided by sending both messages from sensor during the transition, but
+		// that seemed like a lot of extra traffic.  Could have also been simpler of the reconciliationMap
+		// simply used a key instead of taking in a type and using reflection to get the string to use as the key.
+		// Leaving the key defintion to the caller may have simplified this as well as we could have simply used
+		// workerType when adding to the map.  This essentially matches how the sensor side is implemented.
+		workerType = reflectutils.Type(event.Resource)
```

Comment:
I'm not sure I fully understand this portion of the comment and code. 

Using the `SensorEvent_ComplianceOperatorResult` as type for the reconciliation map below I get it. Makes sense. But why aren't you using the `ComplianceOperatorResult` work queue as well? (

i.e.: 
```go
workerType = reflectutils.Type(&central.SensorEvent_ComplianceOperatorResult{}
```

If both messages are going to the same pipeline, what is the need of having two workerQueues for them?

Reply:
TBH, comments like this are the exact reason I wanted review from maple.  That is an excellent point and I will take a look and test out this suggestion.